### ISSUE TITLE
cosmetic issue with the generated nunit output as the test-case name is built with a "." rather than a " "

### ISSUE THREAD

**Issue by Vaevictus** [pester/Pester#1530]
When using the -CI flag, the testreport.xml that is generated is imported fine into azure devops, but the testnames do look strange. The "describe" text and "It" text are concatenated with a . rather than a space

e.g.
 <test-case description="should have managedRuntime set to v4.0" name="The SymbologyServer application pool.should have managedRuntime set to v4.0" t

**Comment by nohwnd** [comment]
Are you reporting this as a regression from v4 or just a general concern? I just looked at the v4 report and it uses the same thing to join the path to the test. Using just a space would make it very difficult to figure out where the Describe description ends and where It description starts. So I don't think this is a regression. But we can talk about ways to do this better in 5.2.

**Comment by Vaevictus** [comment]
Hi, I am reporting this only as a general cosmetic issue and not as a regression, I never tried the CI piece with v4

**Comment by Vaevictus** [comment]
![image](https://user-images.githubusercontent.com/431112/81562427-1d58f200-9395-11ea-8bb1-2e9f8e9bf251.png)

**Comment by nohwnd** [comment]
Gotcha, hopefully when I migrate to Nunit 3 format which has the semantics described better maybe I can find a property that would be better suited for the whole path.

**Comment by fflaten** [comment]
NUnit3 has better attributes:
- name (expanded testname)
- fullname (expanded path incl test)
- classname (path to parent)
- methodname (test name)

The problem though is that AzDo, ReportUnit etc only show the name-attribute, even in the details-view AFAIK. 

So will probably have to use path in name for Nunit3 also unless you want to see "does not throw" on every test without a way to know which block and/or testcase/foreach it was on.

**Comment by nohwnd** [comment]
Yeah that would suck. I tried it as well with azdo and I guess we don't have any other option now than putting the full path in the name, and filing a bug to azdo.

**Comment by fflaten** [comment]
> and filing a bug to azdo. 

@nohwnd Do you know where to report by any chance?

**Comment by nohwnd** [comment]
The `name` attribute is built as `<describe>.<it>` on purpose - the `.` is the path separator NUnit uses to express the suite hierarchy, and the NUnit/ReportGenerator readers and various test explorers rely on it. The `description` attribute already carries just the `It` text (`should have managedRuntime set to v4.0`), which is the human-readable part.

So I'm wary of switching the `.` to a space - that changes the report shape for everyone consuming it. If Azure DevOps is showing the dotted `name` and you'd rather see the plain `It` text, can you say where it's displaying it? There may be a better lever (e.g. surfacing `description`) than changing the separator. Leaving the format as-is for now.
